src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1849,13 +1849,15 @@ namespace {
                 xc < 0 || xc >= (Int) dimensions[3]) {
               continue;
             }
-// gcc 4.6 / 4.7 has a bug here, issues warning -Warray-bounds which is not true, see *), so suppress it
-#if (__GNUC__ == 4 && ( __GNUC_MINOR__ == 6 || __GNUC_MINOR__ == 7) )
+// gcc 4.6 / 4.7 / 5.3 has a bug here, issues warning -Warray-bounds which is not true, see *), so suppress it
+#if ((__GNUC__ == 4 && (__GNUC_MINOR__ == 6 || __GNUC_MINOR__ == 7)) || \
```

Comment:
off-topic: but how about gcc  4.9? That will be used with Ubuntu 16.04 and I recall there were some problems in the past with it.